packages/mongodb-rag-ingest/src/commands/pages.ts

```diff
@@ -7,23 +7,50 @@ const commandModule: CommandModule<
   Record<string, unknown>,
   LoadConfigArgs & PagesCommandArgs
 > = {
-  command: "pages",
+  command: "pages <action>",
   builder(args) {
-    return withConfigOptions(args).option("source", {
-      string: true,
-      description: "A source name to load. If unspecified, loads all sources.",
-    });
+    return args
+      .command({
+        command: "update",
+        describe: "Update pages data from sources",
+        builder: (updateArgs) =>
+          withConfigOptions(updateArgs).option("source", {
+            string: true,
+            description:
+              "A source name to load. If unspecified, loads all sources.",
+          }),
+        handler: (updateArgs) => withConfig(doPagesCommand, updateArgs),
+      })
+      .command({
+        command: "delete [permanent]",
+        describe: "Delete pages data from database",
+        builder: (deleteArgs) =>
+          withConfigOptions(deleteArgs)
+            .option("source", {
+              string: true,
+              description:
+                "A source name to delete. If unspecified, deletes all sources.",
+            })
+            .option("permanent", {
+              boolean: true,
+              description:
+                "If true, permanently deletes the pages. If false or unspecified, marks the pages as deleted without removing them from the collection.",
+            }),
+        handler: (deleteArgs) => withConfig(doDeleteCommand, deleteArgs),
+      })
+      .demandCommand(1, "Specify an action for 'pages' command");
   },
-  handler(args) {
-    return withConfig(doPagesCommand, args);
+  describe: "Manage pages data from sources",
+  handler: (_args) => {
+    console.error('Specify an action for "pages" command');
   },
-  describe: "Update pages data from sources",
 };
 
 export default commandModule;
 
 type PagesCommandArgs = {
```

Comment:
for `PageCommandArgs`, like for embedding, should be at sub-command level, not the root

ie should be `UpdatePagesCommandArgs` and `DeletePagesCommandArgs`